Z/z3/build_tarballs.jl

```diff
@@ -3,42 +3,84 @@
+    ArchiveSource("https://github.com/Gnimuc/JuliaBuilder/releases/download/v1.3.0/julia-1.3.0-x86_64-apple-darwin14.tar.gz", "f2e5359f03314656c06e2a0a28a497f62e78f027dbe7f5155a5710b4914439b1"; unpack_target="julia-x86_64-apple-darwin14"),
+    ArchiveSource("https://github.com/Gnimuc/JuliaBuilder/releases/download/v1.3.0/julia-1.3.0-x86_64-w64-mingw32.tar.gz", "c7b2db68156150d0e882e98e39269301d7bf56660f4fc2e38ed2734a7a8d1551"; unpack_target="julia-x86_64-w64-mingw32"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
+case $target in
+  arm-linux-gnueabihf|x86_64-linux-gnu)
+    Julia_PREFIX=${WORKSPACE}/srcdir/julia-$target/julia-1.3.1
+    Julia_ARGS="-DZ3_BUILD_JULIA_BINDINGS=True -DJulia_PREFIX=${Julia_PREFIX}"
+    ;;
+  x86_64-apple-darwin14|x86_64-w64-mingw32)
+    Julia_PREFIX=${WORKSPACE}/srcdir/julia-$target/juliabin
+    Julia_ARGS="-DZ3_BUILD_JULIA_BINDINGS=True -DJulia_PREFIX=${Julia_PREFIX}"
+    ;;
+esac
+
 cd $WORKSPACE/srcdir/z3/
-atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fix-32-bit-linux.patch
 
 mkdir z3-build && cd z3-build
 cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_FIND_ROOT_PATH="${prefix}" \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    ${Julia_ARGS} \
     ..
 make -j${nproc}
 make install
+install_license ${WORKSPACE}/srcdir/z3/LICENSE.txt
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = expand_cxxstring_abis(supported_platforms())
+platforms_libcxxwrap = [
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Linux(:x86_64, libc=:glibc),
+    MacOS(:x86_64),
+    Windows(:x86_64)
+]
+
+platforms = filter(x->!(x in platforms_libcxxwrap), supported_platforms())
+
+platforms_libcxxwrap = expand_cxxstring_abis(platforms_libcxxwrap)
+platforms = expand_cxxstring_abis(platforms)
 
 # The products that we will ensure are always built
+products_libcxxwrap = [
+    LibraryProduct("libz3", :libz3),
+    LibraryProduct("libz3jl", :libz3jl),
+    ExecutableProduct("z3", :z3)
+]
+
 products = [
     LibraryProduct("libz3", :libz3),
     ExecutableProduct("z3", :z3)
 ]
 
 # Dependencies that must be installed before this package can be built
+dependencies_libcxxwrap = Dependency[
+    Dependency("libcxxwrap_julia_jll")
+]
+
 dependencies = Dependency[
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"8")
+include("../../fancy_toys.jl")
+
+if any(should_build_platform.(triplet.(platforms_libcxxwrap)))
+    build_tarballs(ARGS, name, version, sources, script, platforms_libcxxwrap, products_libcxxwrap, dependencies_libcxxwrap; preferred_gcc_version=v"8")
+end
+if any(should_build_platform.(triplet.(platforms)))
+    build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"8")
+end
```

Comment:
Sorry, I just realised that the `Qt_jll` wasn't the best example, we also need to make sure that `--register` is passed only to the last `build_tarballs` calls.  See [this other example](https://github.com/JuliaPackaging/Yggdrasil/blob/c3e3c4a96c723306b4da23fc6d05f12995b21ed8/G/Git/build_tarballs.jl#L76-L93).  The important bit is to pass `non_reg_ARGS` in the first call and the full `ARGS` in the second and last one.